### PR TITLE
Conversion of graph vignette to RMarkdown file

### PR DESCRIPTION
**Pull request by alatiku**
@Bioconductor/sweave2rmd please review my work. I created the RMarkdown file for the graph vignette.
Attached are the HTML [graph.zip](https://github.com/Bioconductor/graph/files/10943534/graph.zip) and the original PDF [graph.pdf](https://github.com/Bioconductor/graph/files/10943538/graph.pdf) files for comparison.
Thank you.

### CODE REVIEW

**Comment by jwokaty** [comment]
Thank you. The commits are cleaner and it will be easier for me to review this way. I hope you feel more comfortable with git.

- [ ] the `.Rmd` file knits to `HTML`
- [ ] `R CMD build` runs without errors or timeouts
- [ ] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [x] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [ ] any added lines use the same spacing and indents as the existing
  document
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [ ] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [ ] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

File: vignettes/graph.Rmd [hunk 1]
```diff
@@ -0,0 +1,268 @@
+---
+title: "**How to use the graph package**"
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: Remove the `**` around the title

File: vignettes/graph.Rmd [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+author:
+    R. Gentleman [aut],
+    Elizabeth Whalen [aut],<br>
+    W. Huber [aut],
+    S. Falcon [aut],<br>
+    Aliyu Atiku Mustapha [ctb] (Converted 'graph' vignette from Sweave to RMarkdown/HTML.).<br>
+    Bioconductor Package Maintainer [cre] <br>
+        (maintainer@bioconductor.org)
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: This section, lines 1 - 20, should be in YAML. It should also reflect the contents of the [PDF](https://bioconductor.org/packages/3.17/bioc/vignettes/graph/inst/doc/graph.pdf). These authors will be put in the HTML, but they're not on the PDF. You can look at the YAML header in the [clusterGraph vignette](https://github.com/Bioconductor/graph/blob/devel/vignettes/clusterGraph.Rmd)  to see how you might modify the header for graph.Rmd.

File: vignettes/graph.Rmd [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+date: "Last edited `r format(Sys.time(), '%B %d, %Y')`"
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: Remove 'Last edited' since it doesn't show this on the PDF.

File: vignettes/graph.Rmd [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+package: graph
+vignette: >
+    %\VignetteIndexEntry{How to use the graph package}
+    %\VignetteEngine{knitr::rmarkdown}
+    %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document
+
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: Remove this blank line.

File: vignettes/graph.Rmd [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+---
+
+
```
Comment (jwokaty) [inline vignettes/graph.Rmd:26]: Please remove extra blank lines such as line 22 and in the rest of the document.

File: vignettes/graph.Rmd [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+##    Introduction
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: I think this should be 1 `#` rather than 2.

File: vignettes/graph.Rmd [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+The graph package provides an implementation of graphs (the kind with nodes and edges) in R. Software infrastructure is provided by three different, but related packages, <br><br> **graph** Provides the basic class definitions and functionality.<br><br> 
+**RBGL** Provides an interface to graph algorithms (such as shortest path, connectivity etc).<br><br> 
+**Rgraphviz** Provides rendering functionality. Different layout algorithms are provided and node plotting, line type, color etc parameters can be controlled by the user.<br><br> 
+A short description of the R classes and methods is given at the end of this document. But here, we begin by creating some graphs and performing different operations on those graphs.<br>
+The reader will benefit greatly from also having the *Rgraphviz* package available and from using it to
+render the different graphs as they proceed through these notes.
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: Replace `<br><br>` with a blank line. 

Bioconductor packages should have links. Please read [BiocStyle](https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html) to learn how to make canonical links for Bioconductor packages. Do this for `graph`, `RGBL`, and `Rgraphviz`. 

For any other Bioconductor packages, make those links as well.

File: vignettes/graph.Rmd [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+
+##    Getting Started
+
+We will first create a graph and then spend some time examining some of the different functions that can be applied to the graph. We will create a random graph as the basis for our explorations (but will delay explaining the creation of this graph until Section 3).<br>
```
Comment (jwokaty) [inline vignettes/graph.Rmd]: Replace the `<br>` with a blank line.

**Comment by alatiku** [comment]
Hi, I have implemented all the changes you mentioned. I will now add the file, commit and push.
Thank you

File: vignettes/graph.Rmd [hunk 9]
```diff
@@ -0,0 +1,278 @@
+---
+title: "How to use the graph package"
+author: 
+- name: "Aliyu Atiku Mustapha"
+  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML."
+- name: "Bioconductor Package Maintainer"
+  affiliation: maintainer@bioconductor.org
```
Comment (jwokaty) [inline vignettes/graph.Rmd:7]: Remove lines 6-7. The maintainer is required information for the DESCRIPTION file, but the Bioconductor Package Maintainer isn't actually an author or contributor. Typically, the contributor gets added to the `author` and anyone who appears in the original PDF.

File: vignettes/graph.Rmd [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+author: 
+- name: "Aliyu Atiku Mustapha"
+  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML."
+- name: "Bioconductor Package Maintainer"
+  affiliation: maintainer@bioconductor.org    
+date: "`r format(Sys.time(), '%B %d, %Y')`"
+package: graph
+vignette: >
+    %\VignetteEngine{knitr::rmarkdown}
+    %\VignetteIndexEntry{graph}
```
Comment (jwokaty) [inline vignettes/graph.Rmd:12]: Replace `graph` with `How to use the graph package`. The VignetteIndexEntry is generally the same as the title of the document.

File: vignettes/graph.Rmd (hunk 7 above)
Comment (jwokaty) [reply vignettes/graph.Rmd]: The brs have been removed, but it's still not using the recommended way to make links. Please read the [macros section of BiocStyle](https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html#4_Style_macros).

**Pull request closed without merging**